Entity/Helper/JsonClass.php

```diff
@@ -84,10 +85,13 @@ public function jsonSerialize()
         ];
     }
 
-    public function jsonDeserialize()
+    public function jsonDeserialize(object $object = null)
```

Comment:
It is confusing that deserializing an object allows for another object to be it's deserialisation.

I Think you simply need another function that does the deserialisation of the properties when you allready have an object!

`foreach ($this->properties as $name => $value) {
            if (! $reflectionClass->hasProperty($name)) {
                continue;
            }
            $instance->deserialize($name, $value);
        }`